x/revenue/keeper/msg_server.go

```diff
@@ -114,16 +114,15 @@ func (k Keeper) RegisterRevenue(
 		"withdraw", effectiveWithdrawer,
 	)
 
-	ctx.EventManager().EmitEvents(
-		sdk.Events{
-			sdk.NewEvent(
-				types.EventTypeRegisterRevenue,
-				sdk.NewAttribute(sdk.AttributeKeySender, msg.DeployerAddress),
-				sdk.NewAttribute(types.AttributeKeyContract, msg.ContractAddress),
-				sdk.NewAttribute(types.AttributeKeyWithdrawerAddress, effectiveWithdrawer),
-			),
-		},
-	)
+	err := ctx.EventManager().EmitTypedEvent(&types.EventRegisterRevenue{
+		DeployerAddress:     msg.DeployerAddress,
+		ContractAddress:     msg.ContractAddress,
+		EffectiveWithdrawer: effectiveWithdrawer,
+	})
+
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I dont think we should return err here.
The execution of the msg already happened so we would be returning an error of a completed execution.
I think it would be better to should log an error that the event manager failed.